src/zenml/integrations/argilla/annotators/argilla_annotator.py

```diff
@@ -0,0 +1,298 @@
+
+        Returns:
+            The configuration.
+        """
+        return cast(ArgillaAnnotatorConfig, self._config)
+
+    @property
+    def settings_class(self) -> Type[ArgillaAnnotatorSettings]:
+        """Settings class for the Argilla annotator.
+
+        Returns:
+            The settings class.
+        """
+        return ArgillaAnnotatorSettings
+
+    def get_url(self) -> str:
+        """Gets the top-level URL of the annotation interface.
+
+        Returns:
+            The URL of the annotation interface.
+        """
+        return (
+            f"{self.config.instance_url}:{self.config.port}"
+            if self.config.port
+            else self.config.instance_url
+        )
+
+    def _get_client(self) -> ArgillaClient:
+        """Gets Argilla client.
+
+        Returns:
+            Argilla client.
+        """
+        config = self.config
+        init_kwargs = {"api_url": self.get_url()}
+
+        # set the API key from the secret or using settings
+        authentication_secret = self.get_authentication_secret()
+        if config.api_key and authentication_secret:
+            api_key = config.api_key
+            logger.debug(
+                "Both API key and authentication secret are provided. Using API key from settings as priority."
+            )
+        elif authentication_secret:
+            api_key = authentication_secret.secret_values.get("api_key", "")
+            logger.debug("Using API key from secret.")
+        elif config.api_key is not None:
+            api_key = config.api_key
+            logger.debug("Using API key from settings.")
+
+        if api_key:
+            init_kwargs["api_key"] = api_key
+
+        if config.workspace is not None:
+            init_kwargs["workspace"] = config.workspace
+        if config.extra_headers is not None:
+            init_kwargs["extra_headers"] = json.loads(config.extra_headers)
+        if config.httpx_extra_kwargs is not None:
+            init_kwargs["httpx_extra_kwargs"] = json.loads(
+                config.httpx_extra_kwargs
+            )
+
+        try:
+            _ = rg.active_client()
+        except BaseClientError:
+            rg.init(**init_kwargs)
+        return rg.active_client()
+
+    def get_url_for_dataset(self, dataset_name: str) -> str:
+        """Gets the URL of the annotation interface for the given dataset.
+
+        Args:
+            dataset_name: The name of the dataset.
+
+        Returns:
+            The URL of the annotation interface.
+        """
+        dataset_id = self.get_dataset(dataset_name=dataset_name).id
+        return f"{self.get_url()}/dataset/{dataset_id}/annotation-mode"
+
+    def get_datasets(self) -> List[Any]:
+        """Gets the datasets currently available for annotation.
+
+        Returns:
+            A list of datasets.
+        """
+        old_datasets = self._get_client().list_datasets()
+        new_datasets = rg.FeedbackDataset.list()
```

Comment:
Just for my understanding, can you explain the difference between the `old_datasets` and `new_datasets` here?

Reply:
:) Ok this is some Argilla weirdness. So they currently have two different datasets APIs within their package at the same time. They have some old dataset types which use one interface, and they have a new type (the `FeedbackDataset` which has a whole other implementation / interface). So depending on when you started using Argilla, you'll be using one or the other API. (Or both!) They say that they're phasing out the old interface, but probably it's going to be weeks / months before this stabilises, so for now we just have to support both somehow. It's unfortunate.